seed/api/v2_1/views.py

```diff
@@ -120,11 +123,54 @@ def get_queryset(self):
     def building_sync(self, request, pk):
         """
         Return BuildingSync representation of the property
-        ---
 
+        ---
+        parameters:
+            - name: pk
+              description: The PropertyView to return the BuildingSync file
+              type: path
+              required: true
+            - name: organization_id
+              type: integer
+              required: true
+              paramType: query
+            - name: cycle_id
+              type: integer
+              required: true
+              paramType: query
         """
-        return JsonResponse(
-            {"status": "error", "message": "Not yet implemented. PK was {}".format(pk)})
+        # organization_id = request.data['organization_id']
+        cycle_id = request.query_params.get('cycle_id', None)
+
+        if not cycle_id:
+            return JsonResponse({
+                'success': False,
+                'message': "Cycle ID is not defined"
+            })
+        else:
+            cycle = Cycle.objects.get(pk=cycle_id)
+
+        try:
+            # TODO: not checking organization? Is that right?
+            property_view = PropertyView.objects.select_related('state').get(pk=pk, cycle=cycle)
+        except PropertyView.DoesNotExist:
+            return JsonResponse({
+                'success': False,
+                'message': 'Cannot match a PropertyView with pk=%s; cycle_id=%s' % (pk, cycle_id)
+            })
+
+        bs = BuildingSync()
+        # get the latest BSXML file from the PropertyState
+        bs_file = property_view.state.building_files.last().file.path
+        if os.path.exists(bs_file):
+            bs.import_file(bs_file)
+            xml = bs.export(property_view.state, BuildingSync.BRICR_STRUCT)
+            return HttpResponse(xml, content_type='application/xml')
+        else:
+            return JsonResponse({
+                'success': False,
+                'message': "Could not find valid BuildingSync file attached to PropertyState"
+            })
```

Comment:
Walking myself through it; so you:
- get the existing property view by pk
- fetch the last building file for it
- get the path of it
- if it exists in the uploaded files section you 
  - first import the file using the BuildingSync class import_file method
  - get an xml version of it using the BuildingSync export method
  - return that
- if not, just return failure

Now actual comment, what status should we return on failure here? 404?

Reply:
if failure, yeah, 404. I'm going to extend this a bit more today. There are some edge cases that I need to address. Namely:

* Create a new BuildingSync file if an existing one does not exist
* Create new nodes if node is not found
* Handle complex elements (e.g. compound lookups, if FloorAreaType = 'Footprint' then set area to value').